### ISSUE TITLE
Refine chronological display of published notes that were previously saved as drafts

### ISSUE THREAD

**Issue by ebarry** [publiclab/plots2#9985]
**Please describe the desired behavior.**

Returning to a comment from @jywarren from https://github.com/publiclab/plots2/issues/2667#issuecomment-389886458 that was not taken up at that time. It's about updating the date when publishing posts that were drafted, and saved as drafts, on an earlier date: 

> Second, we set a final url with the publish date in it. I wonder if we should reset the url at the moment of final publication, and before then we can use a temporary one, resetting this on publication? This is a little complex but seems workable, no? There's a generate_path method to recalculate the path with the current title and date. That way the title could be changed as well and this would reflect in the final URL.

This would be very helpful as currently if someone publishes a draft even from a week ago, there's too much newer activity for it to even show up on the first page of latest results. 

cc @steviepubliclab

**Comment by jywarren** [comment]
Thanks!

We actually did forward-date the post, but maybe some pages sort by revision timestamp instead of node timestamp? 

https://github.com/publiclab/plots2/blob/a578843b8127c1bf96c5bc023b673a652b0cd074/app/controllers/notes_controller.rb#L413-L420

If so, we should be able to add a couple more lines there. Do you think this is worth writing a test against, or should we just make the additions and see if it works upon publishing?

**Comment by jywarren** [comment]
I think we'd just do this, just after line 419:

```ruby
     @node.latest['timestamp'] = DateTime.now.to_i # odd assignment needed due to legacy Drupal column types 
```

**Comment by jywarren** [comment]
Also, can you double check whether a recently published draft URL matches the publication date/time? According to the quoted code above, we are setting the URL correctly, just not the revision timestamp. But it's possible we have that wrong too, do you have an example of a published draft we can examine to confirm? Thanks!

**Comment by ebarry** [comment]
How about examining this case? 
>there was a community member who posted this on Feb 13, 2020 : https://publiclab.org/notes/WendyBrawer/08-20-2019/how-to-get-street-trees-for-your-community  however it's showed up as an older post a few pages back in the activity feed because she's been drafting it since august 2019.

**Comment by jywarren** [comment]
Thanks, that's very helpful. OK, so we have a few places where sorting happens:

1. on the old (current) dashboard, we sort [by id](https://github.com/publiclab/plots2/blob/a578843b8127c1bf96c5bc023b673a652b0cd074/app/controllers/home_controller.rb#L95) which is a problem because we can't change the unique id. However, with the new dashboard this wouldn't be an issue.
2. at https://publiclab.org/research, which will preserve the dashboard-style "firehose", we also use this display-by-unique-id form. I think we could change this but maybe it's not an issue, @ebarry what do you think?
3. on tag pages, [we sort by revision timestamp](https://github.com/publiclab/plots2/blob/a578843b8127c1bf96c5bc023b673a652b0cd074/app/controllers/tag_controller.rb#L129), as we theorized above, so that should be an easy fix.
4. on the new dashboard, we use topic cards ([template code here](https://github.com/publiclab/plots2/blob/a578843b8127c1bf96c5bc023b673a652b0cd074/app/views/tag/_topicCard.html.erb#L15)) and these are sorted by `node.created` for notes, and `revision.timestamp` for other types (wikis), which would be fixed by the above proposed solution. 

So with the exception of (2) above, this simple fix should solve it, and if you are OK not fixing this on the older dashboard.

Is the https://publiclab.org/research page going to be used enough that we should try to solve it there too?

**Comment by jywarren** [comment]
The minimal fix is now in testing at https://github.com/publiclab/plots2/pull/9989

**Comment by ebarry** [comment]
ah great detective work! Yes, i believe the staff are going to be looking at the firehose view to make sure we're not missing anything, so let's also fix this for /research please. Thank you for sorting this out.

**Comment by jywarren** [comment]
https://github.com/publiclab/plots2/pull/9989 was merged, and if you can test it out on stable that would be great. Fixing it on firehose view is a bit more involved, I'll look into it in the coming week. Thanks!

**Comment by jywarren** [comment]
I think this could involve changing this line to sort by revision.timestamp instead:

https://github.com/publiclab/plots2/blob/a7bcc1761279616203701336e7104df7204b317f/app/controllers/home_controller.rb#L89-L95

But i'm not sure what the ramifications are of that change. It could just work. Or it could cause unexpected issues. I think the best way forward is to just try it out in a pull request. 

@TildaDares would you have any interest in trying this change out? It'd intended to help sort draft notes by their final publication date, rather than their unique id. If not, no worries!

**Comment by TildaDares** [comment]
@jywarren I’ll take a look at it

**Comment by ebarry** [comment]
Just tested on stable.publiclab.org, and it **does not seem to be ordering by date that the `blog` tag was added**. What i did to test: 

- went to https://stable.publiclab.org/blog and observed that the current top post is "Remembering Tonawanda" published on 07-01-2020. 
- went to went to https://stable.publiclab.org/dashboard and scrolled back to find an older note -- https://stable.publiclab.org/notes/sashae/06-26-2020/diy-satellite-ground-station with a posting date 06-26-2020 that pre-dates the Tonawanda note. 
- Added `blog` tag on the "DIY Satellite Ground Station" note
- went to https://stable.publiclab.org/blog, and noticed that the "DIY Satellite Ground Station" note that I just tagged `blog` was not shown above the "Remembering Tonawanda" post.
